src/components/FilmDetails/FilmDetails.js

```diff
@@ -1,15 +1,28 @@
 /* eslint-disable max-len */
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
+import { store } from '../../store';
 
 export class FilmDetails extends Component {
+  componentDidMount() {
+    this.unsubscribe = store.subscribe(() => {
+      this.forceUpdate();
+    });
+  }
+
+  componentWillUnmount() {
+    this.unsubscribe();
+  }
+
   render() {
+    const { match: { params: { id } } } = this.props;
+    const { films } = store.getState();
     const {
       title,
       description,
       imgUrl,
       imdbUrl,
-    } = this.props;
+    } = films.find(item => item.id === id);
```

Comment:
Find film when state is updated, not on each render

Reply:
> Find film when state is updated, not on each render

fixed